mlflow/metrics/genai/genai_metric.py

```diff
@@ -558,4 +604,104 @@ def eval_fn(
         version=version,
         metric_details=evaluation_context["eval_prompt"].__str__(),
         metric_metadata=metric_metadata,
+        genai_metric_args=genai_metric_args,
     )
+
+
+def _filter_by_field(df, field_name, value):
+    return df[df[field_name] == value]
+
+
+def _deserialize_genai_metric_args(args_dict):
+    mlflow_version_at_ser = args_dict.get("mlflow_version")
+    fn_name = args_dict.get("fn_name")
+    if fn_name is None or mlflow_version_at_ser is None:
+        raise MlflowException(
+            message="The artifact JSON file appears to be corrupted and cannot be deserialized. "
+            "Please regenerate the custom metrics and rerun the evaluation. "
+            "Ensure that the file is correctly formatted and not tampered with.",
+            error_code=PUBLIC_UNDOCUMENTED,
+        )
+
+    if mlflow_version_at_ser != VERSION:
+        warnings.warn(
+            f"The custom metric definitions were serialized using MLflow {mlflow_version_at_ser}. "
+            f"Deserializing them with the current version {VERSION} might cause mismatches. "
+            "Please ensure compatibility or consider regenerating the metrics "
+            "using the current version.",
+            UserWarning,
+            stacklevel=2,
+        )
+
+    args_dict.pop("mlflow_version")
+    args_dict.pop("fn_name")
+    if fn_name == make_genai_metric_from_prompt.__name__:
+        return make_genai_metric_from_prompt(**args_dict)
+
+    examples = args_dict["examples"]
+    if examples is None:
+        return make_genai_metric(**args_dict)
+
+    deser_examples = []
+    for example in examples:
+        deser_examples.append(EvaluationExample(**example))
+
+    args_dict["examples"] = deser_examples
+    return make_genai_metric(**args_dict)
+
+
+def search_custom_metrics(
```

Comment:
Do we need a search API that goes through the effort of locally downloading the artifacts, filters them, and only returns one? 
What if we simplified this to return all logged definitions that are part of a given run_id? 
The `search` term makes me feel like this should be a client API, but this is more of a utility fetch function that probably shouldn't be at the top level of the API interface. 
`retrieve_custom_metrics()` or something along those lines seems a bit more relevant (since we're not really search across runs, experiments, or any other logical structure. 
I like the idea of a retrieval function that helps to prevent users from having to use the artifact loading APIs and having to reconstruct the definition from a base config, though! 
I think we could just return a List of the return from the call of `_deserialize_genai_metric_args` and that would be super useful :)

Reply:
^ there was a discussion about this, and the motivation of having `name` and `version` as an additional arguments is that it seems common operation to pick one or two metrics not all logged in the run. This is for convenience, so may not be necessary for first iteration.

Btw I agree with the term `search` may give wrong impression that it provides flexible filtering like other search APIs

Reply:
got it. since the implementation is here I'll keep the name and version.

renaming is to retrieve_custom_metrics to reduce confusion